core/src/main/java/org/apache/any23/writer/BufferedTripleHandler.java

```diff
@@ -0,0 +1,161 @@
+        }
+        ExtractionContext extractionContext;
+        Model extractionModel;
+    }
+
+    private static class WorkflowContext {
+        WorkflowContext(TripleHandler underlying) {
+            this.rootHandler = underlying;
+        }
+
+
+        Stack<String> extractors = new Stack<>();
+        Map<String, ContextHandler> modelMap = new TreeMap<>();
+        IRI documentIRI = null;
+        TripleHandler rootHandler ;
+    }
+
+    public BufferedTripleHandler(TripleHandler underlying) {
+        this.underlying = underlying;
+
+        // hide model in the thread
+        WorkflowContext wc = new WorkflowContext(underlying);
+        BufferedTripleHandler.workflowContext.set(wc);
+    }
+
+    private static final ThreadLocal<WorkflowContext> workflowContext = new ThreadLocal<>();
+
+    /**
+     * Returns model which contains all other models.
+     * @return
+     */
+    public static Model getModel() {
+        return BufferedTripleHandler.workflowContext.get().modelMap.values().stream()
+                .map(ch -> ch.extractionModel)
+                .reduce(new LinkedHashModelFactory().createEmptyModel(), (mf, exm) -> {
+                    mf.addAll(exm);
+                    return mf;
+                });
+    }
+
+    @Override
+    public void startDocument(IRI documentIRI) throws TripleHandlerException {
+        BufferedTripleHandler.workflowContext.get().documentIRI = documentIRI;
+    }
+
+    @Override
+    public void openContext(ExtractionContext context) throws TripleHandlerException {
+        //
+    }
+
+    @Override
+    public void receiveTriple(Resource s, IRI p, Value o, IRI g, ExtractionContext context) throws TripleHandlerException {
+        getModelForContext(context).add(s,p,o,g);
+    }
+
+    @Override
+    public void receiveNamespace(String prefix, String uri, ExtractionContext context) throws TripleHandlerException {
+        getModelForContext(context).setNamespace(prefix, uri);
+    }
+
+    @Override
+    public void closeContext(ExtractionContext context) throws TripleHandlerException {
+        //
+    }
+
+    @Override
+    public void endDocument(IRI documentIRI) throws TripleHandlerException {
+        BufferedTripleHandler.isDocumentFinish = true;
+    }
+
+    @Override
+    public void setContentLength(long contentLength) {
+        underlying.setContentLength(contentLength);
+    }
+
+    @Override
+    public void close() throws TripleHandlerException {
+        underlying.close();
+    }
+
+    /**
+     * Releases content of the model into underlying writer.
+     */
+    public static void releaseModel() throws TripleHandlerException {
+        if(!BufferedTripleHandler.isDocumentFinish) {
+            throw new RuntimeException("Before releasing document should be finished.");
+        }
+
+        WorkflowContext workflowContext = BufferedTripleHandler.workflowContext.get();
+
+        String lastExtractor = ((Stack<String>) workflowContext.extractors).peek();
```

Comment:
Feels hacky... what if not all of the triples came from the same extractor?

Reply:
The idea is to find extractor which produces theustomer's domain-specific rdf graph.

Reply:
@jgrzebyta IMHO, it would be vastly more straightforward to simply have the user extend the [`CompositeTripleHandler`](https://github.com/apache/any23/blob/master/core/src/main/java/org/apache/any23/writer/CompositeTripleHandler.java) class to filter and transform triples into a domain-specific rdf graph of their choosing, before delegating the final domain-specific triple outputs to the wrapped `TripleHandler` instance(s) by calling `super.receiveTriple( [modified triple] )`.

(Analogous in concept to Java's own [`FilterOutputStream`](https://docs.oracle.com/javase/8/docs/api/java/io/FilterOutputStream.html) class.)